extensions/amp-a4a/0.1/a4a-utils.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+import {isArray, isObject} from '../../../src/types';
+import {isSecureUrl} from '../../../src/url';
+import {parseJson} from '../../../src/json';
+
+/** @typedef {{width: number, height: number}} */
+export let SizeInfoDef;
```

Comment:
for future proof, might consider using string to support "100vm"

also, let's call it LayoutInfoDef, so later we can introduce the AMP layout type.